src/Cli.Tests/UtilsTests.cs

```diff
@@ -1,324 +1,325 @@
-        /// and out param for the mergedConfigFile should be null.
-        /// </summary>
-        [DataTestMethod]
-        [DataRow("", false, false, null, false, DisplayName = "If environment value is not set, merged config file is not generated.")]
-        [DataRow("", false, true, null, false, DisplayName = "If environment value is not set, merged config file is not generated.")]
-        [DataRow("", true, false, null, false, DisplayName = "If environment value is not set, merged config file is not generated.")]
-        [DataRow("", true, true, null, false, DisplayName = "If environment value is not set, merged config file is not generated.")]
-        [DataRow(null, false, false, null, false, DisplayName = "If environment variable is removed, merged config file is not generated.")]
-        [DataRow(null, false, true, null, false, DisplayName = "If environment variable is removed, merged config file is not generated.")]
-        [DataRow(null, true, false, null, false, DisplayName = "If environment variable is removed, merged config file is not generated.")]
-        [DataRow(null, true, true, null, false, DisplayName = "If environment variable is removed, merged config file is not generated.")]
-        [DataRow("Test", false, false, null, false, DisplayName = "Environment value set but base config not available, merged config file is not generated.")]
-        [DataRow("Test", false, true, null, false, DisplayName = "Environment value set but base config not available, merged config file is not generated.")]
-        [DataRow("Test", true, false, null, false, DisplayName = "Environment value set but env based config not available, merged config file is not generated.")]
-        [DataRow("Test", true, true, "dab-config.Test.merged.json", true, DisplayName = "Environment value set and both base and envConfig available, merged config file is generated.")]
-        public void TestMergeConfigAvailability(
-            string? environmentValue,
-            bool isBaseConfigPresent,
-            bool isEnvironmentBasedConfigPresent,
-            string? expectedMergedConfigFileName,
-            bool expectedIsMergedConfigAvailable)
+        // Setting up the test scenarios
+        Environment.SetEnvironmentVariable(RuntimeConfigLoader.RUNTIME_ENVIRONMENT_VAR_NAME, environmentValue);
+        string baseConfig = "dab-config.json";
+        string envBasedConfig = "dab-config.Test.json";
+
+        if (isBaseConfigPresent)
         {
-            // Setting up the test scenarios
-            Environment.SetEnvironmentVariable(RUNTIME_ENVIRONMENT_VAR_NAME, environmentValue);
-            string baseConfig = "dab-config.json";
-            string envBasedConfig = "dab-config.Test.json";
-            if (File.Exists(baseConfig))
-            {
-                File.Delete(baseConfig);
-            }
-
-            if (File.Exists(envBasedConfig))
-            {
-                File.Delete(envBasedConfig);
-            }
```

Comment:
Does removing the code to cleanup files created by other tests result in false positive/negative results here? Or is the fact that you're now using a MockFileSystem result in a clean slate for each test case?

Reply:
The advantage of using the mock file system is that it resides entirely in memory, so as the object is GC'ed at the end of the test, the "files" are also removed from memory, meaning we don't have to do our own cleanup.

It also means we can do more parallel tests as each test is isolated from the others because it's just memory objects.